packages/data/addon/builder/requestV2.ts

```diff
@@ -0,0 +1,176 @@
+      type,
+      field,
+      parameters,
+      ...(alias ? { alias } : null),
+      cid: cid ?? nanoid(10)
+    });
+    return this;
+  }
+
+  /**
+   * Convenience for adding a metric column
+   * @param metricName Name of metric
+   * @param parameters Dictionary of parameters
+   */
+  metric(metricName: string, parameters: Parameters = {}): RequestV2Builder {
+    this.column('metric', metricName, parameters);
+    return this;
+  }
+
+  /**
+   * Convenience for adding a dimension column
+   * @param dimensionName Name of dimension
+   * @param parameters Dictionary of parameters
+   */
+  dimension(dimensionName: string, parameters: Parameters = {}): RequestV2Builder {
+    this.column('dimension', dimensionName, parameters);
+    return this;
+  }
+
+  /**
+   * Adds a filter to the request
+   * @param type metric, dimension, or timeDimension
+   * @param field field name
+   * @param parameters dictionary of parameters
+   * @param operator operator
+   * @param values array of values for the filter
+   */
+  filter(
+    type: ColumnType,
+    field: string,
+    parameters: Parameters,
+    operator: FilterOperator,
+    values: (string | number | boolean)[]
+  ): RequestV2Builder {
+    this.request.filters.push({
+      type,
+      field,
+      parameters,
+      operator,
+      values
+    });
+    return this;
+  }
+
+  /**
+   * Adds a row limit to request
+   * @param limit number of max rows
+   */
+  limit(limit: number): RequestV2Builder {
+    this.request.limit = limit;
+    return this;
+  }
+
+  /**
+   * Adds sort to the request
+   * @param type metric,dimension or timeDimension
+   * @param field field name to sort by
+   * @param parameters Dictionary of parameters
+   * @param direction asc/desc
+   */
+  sort(type: ColumnType, field: string, parameters: Parameters, direction: SortDirection): RequestV2Builder {
+    this.request.sorts.push({
+      type,
+      field,
+      parameters,
+      direction
+    });
+    return this;
+  }
+
+  /**
+   * Adds table name to request
+   * @param name name of the table
+   */
+  table(name: string): RequestV2Builder {
+    this.request.table = name;
+    return this;
+  }
+
+  /**
+   * Allows specifies the datasource
+   * @param dataSource name of the datasource
+   */
+  dataSource(dataSource: string): RequestV2Builder {
+    this.request.dataSource = dataSource;
+    return this;
+  }
+
+  /**
+   * Convenience method for adding an interval filter
+   * @param grain 'day','week','month' etc... whatever your datasource supports
+   * @param start Period, datestring or macro
+   * @param end Period, datestring or macro
+   */
+  interval(grain: string, start: string, end: string): RequestV2Builder {
+    const table = this.request.table;
+    assert('Set the table before you set the interval', table.length !== 0);
+
+    this.column('timeDimension', `${table}.dateTime`, { grain }).filter(
+      'timeDimension',
+      `${table}.dateTime`,
```

Comment:
This is only really a Fili/Bard thing. Ideally, all the methods would take a column `id`, whatever it maybe. For Fili/Bard, the `id` for a timeDimension would be `tableA.dateTime`.

I guess this logic would make sense if this was a BardRequest(Builder).

Reply:
There isn't a concept of a Bard Request V2 and Elide Request V2, and probably better off that way.  I think the idea is that we do this particular thing enough in Bard/Fili that we really need a convenience method for this to make the builder worth it.  Those that are making an elide request without a date time filter simply not use this, or use raw addColumn/addFilter methods.  If there is a way to make this work conveniently for both I'm open to it. 

I could simply just add an optional column id to use here, and default to tableA.dateTime, Or I could add an option switch. Or I could just make this: `addBardInterval`.

Reply:
> There isn't a concept of a Bard Request V2 and Elide Request V2

Correct, but this is a higher level object than the RequestV2 interface that could conveniently normalize down to the required interface. However, this would not be my first choice 😄 

>  Those that are making an elide request without a date time filter simply not use this

If this is a general request object, I don't think we want a method that only works on specific data sources


I think the simplest approach would be to remove this logic and require users to pass the `id` as is. I agree it loses some convenience for bard users, but I think it is the most straight forward. Especially when the response column will be in the same form (`{table}.{timeDimension}`)

Reply:
sure that's fine.